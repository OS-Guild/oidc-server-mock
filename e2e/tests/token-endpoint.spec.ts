import * as querystring from 'querystring';
import * as dotenv from 'dotenv';
import axios from 'axios';
import { decode } from 'jws';

import clients from '../config/clients-configuration.json';
import type { Client } from '../types';

describe('Token Endpoint', () => {
  let client: Client;

  beforeAll(() => {
    dotenv.config();
    client = clients.find(c => c.ClientId === 'client-credentials-mock-client-id');
    expect(client).toBeDefined();
  });

  test('Client Credentials', async () => {
    const parameters = {
      client_id: client.ClientId,
      client_secret: client.ClientSecrets?.[0],
      grant_type: 'client_credentials',
      scope: client.AllowedScopes[0],
    };

    const response = await axios.post(process.env.OIDC_TOKEN_URL, querystring.stringify(parameters));

    expect(response).toBeDefined();
    expect(response.data.access_token).toBeDefined();
    const token = decode(response.data.access_token);

    expect(token).toMatchSnapshot();
  });
});
