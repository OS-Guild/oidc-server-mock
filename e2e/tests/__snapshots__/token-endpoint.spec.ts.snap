// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Token Endpoint Client Credentials 1`] = `
{
  "alg": "RS256",
  "typ": "JWT",
  "iss": "http://localhost:8080",
  "aud": "some-app",
  "client_id": "client-credentials-mock-client-id",
  "string_claim": "string_claim_value",
  "scope": [
    "some-app-scope-1"
  ],
  "json_claim": [
    "value1",
    "value2"
  ]
}
`;
